contributors/design-proposals/node/troubleshoot-running-pods.md

```diff
@@ -106,13 +101,88 @@ subsequently be used to reattach and is reported by `kubectl describe`.
+        metav1.TypeMeta `json:",inline"`
+
+        // Spec describes the Ephemeral Container to be created.
+        Spec *Container `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+        // Most recently observed status of the container.
+        // This data may not be up to date.
+        // Populated by the system.
+        // Read-only.
+        // +optional
+        Status *ContainerStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+
+        // If set, the name of the container from PodSpec that this ephemeral container targets.
+        // If not set then the ephemeral container is run in whatever namespaces are shared
+        // for the pod.
+        TargetContainerName string `json:"targetContainerName,omitempty" protobuf:"bytes,4,opt,name=targetContainerName"`
+}
+```
+
+All Ephemeral Containers that have been created in a pod are listed in the pod's
+status:
+
+```
+type PodStatus struct {
+        ...
+        // List of user-initiated ephemeral containers that have been run in this pod.
+        // +optional
+        EphemeralContainers []EphemeralContainer `json:"commands,omitempty" protobuf:"bytes,11,rep,name=ephemeralContainers"`
+
+}
+```
 
-We will extend `v1.Pod`'s `/exec` subresource to support "executing" container
-images. The current `/exec` endpoint must implement `GET` to support streaming
-for all clients. We don't want to encode a (potentially large) `v1.Container` as
-an HTTP parameter, so we must extend `v1.PodExecOptions` with the specific
-fields required for creating a Debug Container:
+To create a new Ephemeral Container, one appends a new `EphemeralContainer` with
+the desired `v1.Container` as `Spec` in `Pod.Status` and updates the `Pod` in
+the API. This is accomplished via a new subresource, `/ephemeralcontainers`,
+which enforces the append-only semantics and authorization. This is similar to
+the `/status` subresource used by the kubelet to modify pod status.
+
+**Note that Ephemeral Containers are not regular containers and should not be
+used to build services.** They lack guarantees for resources or execution, and
+many of the fields of `v1.Container` will not be allowed for Debug Containers. A
+pod update will fail validation if any field is set other than the following
+whitelisted fields: `Name`, `Image`, `Command`, `Args`, `WorkingDir`, `Env`,
+`EnvFrom`, `ImagePullPolicy`, `SecurityContext`. `TTY` and `Stdin` are always
+enabled for Debug Containers and will be ignored.
+
+Once the pod object is updated, the kubelet worker watching this pod will launch
+the Ephemeral Container and update its status. The client is expected to watch
+for the creation of the container status and then attach to the console of a
+debug container using the existing attach endpoint,
+`/api/v1/namespaces/$NS/pods/$POD_NAME/attach`. Note that any output of the new
+container between its creation and subsequent attach will not be replayed and
+can only be viewed using `kubectl log`.
```

Comment:
Does this mean `kubectl log` will be supported for debug containers and is it treated as a regular container? What other commands can be performed against a debug container, e.g., `exec`? I think it's worth calling out what operations will be supported for debug containers.